dandiapi/api/views/dandiset.py

```diff
@@ -453,3 +463,33 @@ def users(self, request, dandiset__pk):  # noqa: C901
                 )
 
         return Response(owners, status=status.HTTP_200_OK)
+
+    @swagger_auto_schema(
+        methods=['GET'],
+        manual_parameters=[DANDISET_PK_PARAM],
+        request_body=no_body,
+        operation_summary='List active/incomplete uploads in this dandiset.',
+    )
+    @action(methods=['GET'], detail=True)
+    def uploads(self, request, dandiset__pk):
+        dandiset: Dandiset = self.get_object()
+
+        # Special case where a "safe" method is access restricted, due to the nature of uploads
+        self.require_owner_perm(dandiset)
+
+        uploads: QuerySet[Upload] = dandiset.uploads.all()
+        serializer = DandisetUploadSerializer(instance=uploads, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
```

Comment:
I think we should paginate these. Someone could theoretically DoS the server by creating thousands of uploads (which may be a completely legitimate action) and then hitting this endpoint.

Reply:
Just pushed that change. It's a bit awkward since we can't use the default pagination behavior, due to this being a "detailed" endpoint that still needs pagination. That was originally the reason I didn't bother, since it seemed unlikely to be an issue anyway, but better safe than sorry. I've also changed the client to match that, and tested it locally, confirming the overall behavior is unchanged.